containers/src/Kokkos_ScatterView.hpp

```diff
@@ -217,8 +217,11 @@ struct ScatterValue<ValueType, Kokkos::Experimental::ScatterSum, DeviceType,
  public:
   KOKKOS_FORCEINLINE_FUNCTION ScatterValue(ValueType& value_in)
       : value(value_in) {}
-  KOKKOS_FORCEINLINE_FUNCTION ScatterValue(ScatterValue&& other)
-      : value(other.value) {}
+  KOKKOS_FUNCTION ScatterValue(const ScatterValue&)                 = delete;
+  KOKKOS_DEFAULTED_FUNCTION ScatterValue(ScatterValue&&)            = default;
+  KOKKOS_FUNCTION ScatterValue& operator=(const ScatterValue&)      = delete;
+  KOKKOS_DEFAULTED_FUNCTION ScatterValue& operator=(ScatterValue&&) = default;
+  KOKKOS_DEFAULTED_FUNCTION ~ScatterValue()                         = default;
```

Comment:
Why prefer this to the rule of zero here?
The reference data member cannot rebind so that is implicit.

Same comment everywhere below.